### DISCUSSION TITLE
Automate removing properties files

### DISCUSSION THREAD

**Discussion by xjim1** [OpenSageTV/Community#173] [General]
Ever since I think I installed Phoenix, or Gemstone, or some other addon, every recording now gets a .properties file.  I regularly go in and delete them, but would like to automate it.
I'm not looking for anyone to share any programs of their own, but would like to learn some simple programming tools to accomplish it.

I've had several years experience scripting using cli's on mainframes from decades past, and know the basics of higher level languages such as c++, Pascal, etc, but have never tried anything on a pc.  I'd prefer a simple scripting language(Python?) that I can learn to purge these files and maybe automate other things.

Thoughts and input welcome.

**Comment by LoisGNS** [comment]
I don't think you really need a full-fledged programming language for this, though you could use one to make it more elegant than the way I do it:

I use a batch file; currently I'm saving my .properties and .my files (both of which get generated) to a separate location, so I use a "move" command for each location where they might be, for example:

move "[source location]\*.properties" [destination]
move "[source location]\*.my" [destination]
move "[another source location]\*.properties" [destination]
etc.

If I just wanted to delete them it would be

del "[source location]\*.properties" 

I have a number of locations, and it would be more elegant if I could iterate through all the subfolders in my imported video folder (and/or others where that might be helpful), but currently I've just entered the most-used ones into my batch file by hand.

I don't run this on a schedule; just whenever I feel like doing a cleanup.

**Comment by KeithAbb0tt** [comment]
I use a php script on my Unraid server.  The script archives all associated media files (.edl, .log, .properties, .txt, etc.) that do not have a corresponding recording file.  The script archives any files found to a directory that SageTV is not aware of.  If a file gets archived that I later find out that I need, I can just copy it back from the archive directory.  I've been running this nightly for over 1 1/2 years now, and I have not needed to un-archive any files yet.

**Comment by xjim1** [comment]
I thought about just using dos commands but I tried using it several years ago probably under XP, but it had its limitations.  I may still do that for a short term thing, but I want it to be a little more extensible if I want to grow what I do.  I'm looking again at PowerShell but don't know enough about it to know if it'll be a good fit.

I've never thought of using php.  Always thought it was a browser script, but looks like I don't have to use it that way?

I want it to be able to do some decision making, query the user, basic error handling and logging.  That's about all the further I've taken it.  
Though I think a dos batch job would be a good start, I'm kinda interested in learning something new at the same time.

**Comment by KeithAbb0tt** [comment]
You might want to take a look at AutoIt, a scripting language strictly for Windows.  It hasn't been updated for a few years, but also probably hasn't really needed to be updated.  One of the cool things about it is the script can be compiled down to a single .exe file, or the script can be executed via the interpreter.

**Comment by jusjoken** [comment]
It's been awhile since I tried this but doesn't BMT have this functionality
built into the tools menu...to remove all properties files?  You may want
to check it out.

Also, there is a sage property itself that disables the creation of
property files but I cannot recall which one.  I will look when I get a
chance but disabling the creation would be the best option

k

**Comment by xjim1** [comment]
I think I tried BMT once, and if iirc, couldn't figure it out to save my life.  Of course, I didn't rtfm, but can usually figure out a web interface.  Not that one.  

I did a quick search thru sage.properties and didn't see anything obvious, but I may not know what I'm looking for.  I agree an internal solution is best and any help there is appreciated as well.  And I just want to delete .properties files created for recordings.  I import a lot of videos and really depend on those getting automatically generated.

I started looking at AutoIt and it may be the answer, for use even outside of Sage.  It can deal with arrays and I've always wanted a little more control parsing spreadsheets and such. A bit of a learning curve, but it seems to have an active forum and lots of coding examples.

**Comment by JustFred51** [comment]
When I noticed that .properties files were being created for every recording, a bit of sleuthing determined that (long-forgotten) BMT was automatically creating them.  Since I never really warmed to using BMT anyway, removing the BMT plugin was the solution to the growth of .properties files.  After that, a one-time manual deletion of the files was easy.

**Comment by xjim1** [comment]
Well, I do have BMT installed and don't use it.  Can these(ok, this) plugin still be reinstalled?  A little worried it may also be responsible for creating them for my imported videos and don't want to break that.

**Comment by JustFred51** [comment]
BMT still shows up in Sage's list of available plugins, so it should be able to be re-installed.

**Comment by eelpout** [comment]
didn't one of the commercial skipping plugins have this feature, to remove **all** associated files of deleted recordings? Maybe it used BMT as a backend for this, not sure.

**Reply by wayner9** [reply]
SJQ had a script to this if I recall.

**Comment by xjim1** [comment]
I found this in the Phoenix settings.  I imagine it's what also creates the metadata for my imported videos, and I want to keep those.
![BMT Metadata Snapshot](https://github.com/user-attachments/assets/71bfc1b2-8fbc-4489-b022-baf1459df802)

**Comment by jusjoken** [comment]
Correct.   That is what allows but to get Metadata for all media including
recordings
